core/core.go

```diff
@@ -0,0 +1,90 @@
+package core
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+	"time"
+
+	"github.com/mitre/gocat/agent"
+	"github.com/mitre/gocat/output"
+
+	_ "github.com/mitre/gocat/execute/shells" // necessary to initialize all submodules
+)
+
+// Initializes and returns sandcat agent.
+func initializeCore(server string, group string,c2 map[string]string, p2pReceiversOn bool, verbose bool) (*agent.Agent, error) {
+	output.SetVerbose(verbose)
+	output.VerbosePrint("Starting sandcat in verbose mode.")
+	return agent.AgentFactory(server, group, c2, p2pReceiversOn)
+}
+
+//Core is the main function as wrapped by sandcat.go
+func Core(server string, group string, delay int, c2 map[string]string, p2pReceiversOn bool, verbose bool) {
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
```

Comment:
We want to be able to easily add rotating SSL/TLS.  Let's set this in either the `Agent` when you're initializing contact interfaces or in the contact interface itself. If there is not x509 certificate available in a string, we will default to this.

Reply:
I'm thinking we can put it in the contact interface, since not all contact implementations will use HTTP(s). Maybe we can add another method to the interface, something like `InitializeContactChannel(profile map[string]interface{}, configuration map[string]interface{}) error`?